app/Http/Controllers/ProductController.php

```diff
@@ -2,23 +2,60 @@
 
 namespace App\Http\Controllers;
 
-use Illuminate\Http\Request;
-
+//use Illuminate\Http\Requestsiuse App\Repositories\ProductRepository as Product;
 use App\Http\Requests;
-
+use App\Http\Controllers\Controller;
 use App\Models\Product;
+use App\Models\Category;
+//use Request;
+use Illuminate\Http\Request;
+use App\Http\Requests\ProductRequest;
+use DB;
 
 class ProductController extends Controller
 {
     /**
-     * Display a listing of the resource.
+     * Show the form for creating a new resource.
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function create()
+    {
+        $categories = Category::all();
+        return view('product.create', compact('categories'));
+    }
+    /**
+     * Create a new product instance.
+     *
+     * @param Request $request request
+     *
+     * @return \Illuminate\Http\Response
+    */
+    public function store(ProductRequest $request)
     {
-        $products = Product::all();
+        try {
+             $product = new Product;
```

Comment:
[rank C] please use create form of Laravel, we don't need to assign one by one for it.